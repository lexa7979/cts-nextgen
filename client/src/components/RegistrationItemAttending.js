/* eslint-disable react/prop-types */
/* eslint-disable no-use-before-define */

import React from "react";
import * as yup from "yup";

import Grid from "@material-ui/core/Grid";
import FormLabel from "@material-ui/core/FormLabel";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Radio from "@material-ui/core/Radio";

const propsSchema = yup.object( {
	formikBag: yup.object().required(),
} );

export default RegistrationItemAttending;

/**
 * @param	{object}	props
 *
 * @returns	{object}	React component
 */
function RegistrationItemAttending( props ) {
	propsSchema.strict().validate( props );

	const { values, handleChange } = props.formikBag;

	return (
		<Grid item container direction="column" justify="center" alignItems="stretch">
			<Grid item xs={12}>
				<FormLabel component="legend">
					Are you attending the conference?
				</FormLabel>
			</Grid>
			<Grid item xs={12}>
				<RadioGroup
					aria-label="position"
					name="attending"
					value={values.attending}
					onChange={handleChange}
					row
				>
					<FormControlLabel
						value="yes"
						label="Yes"
						labelPlacement="bottom"
						control={<Radio color="primary"/>}
					/>
					<FormControlLabel
						value="no"
						label="No"
						labelPlacement="bottom"
						control={<Radio color="primary"/>}
					/>
					<FormControlLabel
						value="maybe"
						label="Maybe"
						labelPlacement="bottom"
						control={<Radio color="primary"/>}
					/>
				</RadioGroup>
			</Grid>
		</Grid>
	);
}
