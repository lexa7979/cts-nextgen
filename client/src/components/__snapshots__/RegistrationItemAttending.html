
<!DOCTYPE html>
<html style="height: 100%;">
  <head>
  </head>
  <body style="display: flex; flex-flow: column nowrap; justify-content: center; align-items: center; height: 100%;">
    <h2>
      Component RegistrationItemAttending - when rendering into snapshots with valid properties - delivers expected result  (-> check snapshot)
    </h2>
    <!--[ Content from unit test: ]-->
    <div class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center">
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12">
        <legend class="MuiFormLabel-root">
          Are you attending the conference?
        </legend>
      </div>
      <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12">
        <div class="MuiFormGroup-root MuiFormGroup-row"
             role="radiogroup"
             aria-label="choose one"
        >
          <label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom">
            <span class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-305 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-checked-306 Mui-checked MuiIconButton-colorPrimary"
                  aria-disabled="false"
            >
              <span class="MuiIconButton-label">
                <input type="radio"
                       checked
                       class="PrivateSwitchBase-input-308"
                       name="attending"
                       value="yes"
                >
                <div class="PrivateRadioButtonIcon-root-321 PrivateRadioButtonIcon-checked-323">
                  <svg class="MuiSvgIcon-root"
                       focusable="false"
                       viewbox="0 0 24 24"
                       aria-hidden="true"
                       role="presentation"
                  >
                    <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z">
                    </path>
                  </svg>
                  <svg class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-322"
                       focusable="false"
                       viewbox="0 0 24 24"
                       aria-hidden="true"
                       role="presentation"
                  >
                    <path d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z">
                    </path>
                  </svg>
                </div>
              </span>
            </span>
            <span class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1">
              Yes
            </span>
          </label>
          <label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom">
            <span class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-305 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
                  aria-disabled="false"
            >
              <span class="MuiIconButton-label">
                <input type="radio"
                       class="PrivateSwitchBase-input-308"
                       name="attending"
                       value="no"
                >
                <div class="PrivateRadioButtonIcon-root-321">
                  <svg class="MuiSvgIcon-root"
                       focusable="false"
                       viewbox="0 0 24 24"
                       aria-hidden="true"
                       role="presentation"
                  >
                    <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z">
                    </path>
                  </svg>
                  <svg class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-322"
                       focusable="false"
                       viewbox="0 0 24 24"
                       aria-hidden="true"
                       role="presentation"
                  >
                    <path d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z">
                    </path>
                  </svg>
                </div>
              </span>
            </span>
            <span class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1">
              No
            </span>
          </label>
          <label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementBottom">
            <span class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-305 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
                  aria-disabled="false"
            >
              <span class="MuiIconButton-label">
                <input type="radio"
                       class="PrivateSwitchBase-input-308"
                       name="attending"
                       value="maybe"
                >
                <div class="PrivateRadioButtonIcon-root-321">
                  <svg class="MuiSvgIcon-root"
                       focusable="false"
                       viewbox="0 0 24 24"
                       aria-hidden="true"
                       role="presentation"
                  >
                    <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z">
                    </path>
                  </svg>
                  <svg class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-322"
                       focusable="false"
                       viewbox="0 0 24 24"
                       aria-hidden="true"
                       role="presentation"
                  >
                    <path d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z">
                    </path>
                  </svg>
                </div>
              </span>
            </span>
            <span class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1">
              Maybe
            </span>
          </label>
        </div>
      </div>
    </div>
    <!--[ End of included content ]-->
    <br>
    <br>
    &lt;RegistrationItemAttending formikBag={exampleFormikBag} /&gt;
  </body>
</html>
